pandas/tests/sparse/test_series.py

```diff
@@ -88,6 +88,18 @@ def setup_method(self, method):
         self.ziseries2 = SparseSeries(arr, index=index, kind='integer',
                                       fill_value=0)
 
+    def test_constructor_data_input(self):
+        # see gh-16905
+        constructor_dict = {1: 1}
+        index = [0, 1, 2]
+        series = pd.Series(constructor_dict, index=index)
+
+        result = SparseSeries(constructor_dict)
+        tm.assert_dict_equal(result.to_dict(), constructor_dict)
+
+        result = SparseSeries(series, index=index)
+        tm.assert_series_equal(result, series, check_series_type=False)
+
```

Comment:
Okay : in light of what you just said, I think you can condense this test a bit and rename as follows:

~~~python
def test_constructor_dict_input(self):
...
series = pd.Series(constructor_dict, index=index)
expected = SparseSeries(series, index=index)

result = SparseSeries(constructor_dict)
tm.assert_sp_series_equal(result, expected)
~~~

And delete everything else below it.